src/Messages/Interactions/TSMessage.m

```diff
@@ -104,7 +111,12 @@ - (instancetype)initWithTimestamp:(uint64_t)timestamp
     _expiresInSeconds = expiresInSeconds;
     _expireStartedAt = expireStartedAt;
     [self updateExpiresAt];
-    _receivedAtDate = [NSDate date];
+
+    if ([self shouldUseReceiptDateForSorting]) {
```

Comment:
These conditions read backwards. Isn't "now" (aka `[NSDate ows_millisecondTimeStamp];` or `[NSDate new]` the receipt date?